CI: Publish Release fails — mcp-publisher_*_linux_amd64.tar.gz not found

## Context

The **Publish Release** workflow (`mcp-registry` job) is failing on every release tag with:

```
tar (child): mcp-publisher_*_linux_amd64.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
##[error]Process completed with exit code 2
```

Most recent failure: run `24932945723` on tag `v1.2.0`.

## Likely root cause

The `tar` step expects a glob `mcp-publisher_*_linux_amd64.tar.gz` to match a file already on disk. Because the glob isn't matching, either:

- The download step earlier in the job silently succeeded with the wrong filename (upstream asset naming changed), or
- The download step failed but didn't `set -e`, so the workflow continued to the `tar` and crashed there.

The upstream project is [modelcontextprotocol/registry](https://github.com/modelcontextprotocol/registry); its release artifacts may have been renamed since this workflow was written.

## Acceptance criteria

- [ ] Inspect `.github/workflows/publish.yml` — find the step that downloads the `mcp-publisher` archive (probably `curl` or `wget` with a wildcard or templated filename).
- [ ] Verify the actual current asset name on the latest [`modelcontextprotocol/registry` release](https://github.com/modelcontextprotocol/registry/releases) and update the download URL / glob pattern accordingly.
- [ ] Add `set -euo pipefail` to the script step so future asset-name changes fail fast at the download instead of cascading into the tar step.
- [ ] Trigger a test re-run of the Publish Release workflow (re-run the failed `v1.2.0` job, or cut a no-op patch tag) to confirm the fix.

## Files involved

- `.github/workflows/publish.yml` (mcp-registry job)

## Effort

XS (~5-15 LOC, mostly URL/glob string update + a defensive `set -e`). Largest cost is verifying the current upstream naming convention.

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -38,9 +38,10 @@ jobs:
         env:
           GH_TOKEN: ${{ github.token }}
         run: |
+          set -euo pipefail
           gh release download --repo modelcontextprotocol/registry \
-            --pattern '*linux_amd64.tar.gz'
-          tar xzf mcp-publisher_*_linux_amd64.tar.gz
+            --pattern 'mcp-publisher_linux_amd64.tar.gz'
+          tar xzf mcp-publisher_linux_amd64.tar.gz
           chmod +x mcp-publisher
 
       - name: Publish to MCP Registry

